src/Freya.Core/Integration.fs

```diff
@@ -25,24 +25,47 @@ open System.Threading.Tasks
 
 (* OWIN Types *)
 
+/// Type alias of <see cref="FreyaEnvironment" /> in terms of OWIN.
 type OwinEnvironment =
     FreyaEnvironment
 
+/// Type alias for the F# equivalent of the OWIN AppFunc signature.
 type OwinApp = 
     OwinEnvironment -> Async<unit>
 
+/// Type alias for the OWIN AppFunc signature.
 type OwinAppFunc = 
     Func<OwinEnvironment, Task>
 
 (* OWIN Conversion *)
 
+/// Provides transformation functions for converting to/from OWIN from/to Freya.
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation (CompilationRepresentationFlags.ModuleSuffix)>]
 module OwinAppFunc =
+    open Freya.Core.Functions
 
+    /// Converts a <see cref="Freya{T}" /> computation to an <see cref="OwinAppFunc" />.
     [<CompiledName ("FromFreya")>]
     let fromFreya (freya: Freya<_>) =
-        OwinAppFunc (fun e -> 
-            Async.StartAsTask (async { 
+        OwinAppFunc (fun e ->
+            async {
                 do! freya { Environment = e
-                            Meta = { Memos = Map.empty } } |> Async.Ignore }) :> Task)
\ No newline at end of file
+                            Meta = { Memos = Map.empty } } |> Async.Ignore }
+            |> Async.StartAsTask :> Task)
+    
+    /// Converts an <see cref="OwinAppFunc" /> to a <see cref="Freya{T}" /> computation
+    /// to allow use of standard OWIN components within Freya.
+    /// NOTE: EXPERIMENTAL
+    [<CompiledName ("ToFreya")>]
+    let toFreya (app: OwinAppFunc) : Freya<unit> =
```

Comment:
I thought you might like this reversal function. Looks like this should work for integrating with existing OWIN apps and middleware. See the tests below that show going back and forth.